engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/execution/JobExecutionEngine.scala

```diff
@@ -50,12 +50,13 @@ final class JobExecutionEngine(
     context.newCachedThreadPool(
       "prioritized-job-pool",
       2,
-      Integer.MAX_VALUE,
+      4,
```

Comment:
I don't like the idea of the pool growing infinitely. Behind the scenes _cached_ threadpools  were assuming _infinite_ number of threads and a zero size blocking queue. That, I believe, wasn't correct as it would easily hide problems in execution without making it visible. I believe it's better to have a fixed maximal size of threadpool and queue of tasks that can grow up to some point.
If the queue of tasks gets full, I picked some reasonable constant I hope, then it will apply the default policy (that is rejecting tasks). At least that way we will know that something is wrong. Either way at that point it is likely a bigger, non-recoverable, problem happened.

Reply:
> don't like the idea of the pool growing infinitely.

OK to adding _upper limit_ to _unlimited pools_.